freestyle-logging/shared/src/main/scala/logging.scala

```diff
@@ -21,21 +21,21 @@ object logging {
   @free
   trait LoggingM {
 
-    def debug(msg: String): FS[Unit]
+    def debug(msg: String, sourceAndLineInfo: Boolean = false)(implicit line: sourcecode.Line, file: sourcecode.File): FS[Unit]
```

Comment:
A comment out of context of this PR.

Because the `msg` parameter is pass by value, there will be a performance hit(string concatenation) if the log level is high/low enough. `verizon/journal` deals with this by having a macro guard the actual `logger.debug` call guarded by adding "if logLevel.isEnabled". Is there a way to do that with the `LoggingM` algebra or to "automatically" lift the `msg` parameter to a `Function0`?